pkg/nuctl/command/invoke.go

```diff
@@ -151,6 +173,41 @@ func newInvokeCommandeer(ctx context.Context, rootCommandeer *RootCommandeer) *i
 	return commandeer
 }
 
+func (i *invokeCommandeer) enrichOptionsForExternalIP(invocationURLs []string) error {
+
+	// provided external ip address,
+	if i.externalIPAddresses != "" {
+
+		// function has node port
+		if functionNodePort := i.createFunctionInvocationOptions.
+			FunctionInstance.
+			GetStatus().
+			HTTPPort; functionNodePort != 0 {
+			externalIPAddresses, err := i.rootCommandeer.platform.GetExternalIPAddresses()
+			if err != nil {
+				return errors.Wrap(err, "Failed to get external IP addresses")
+			}
+			i.createFunctionInvocationOptions.URL = fmt.Sprintf("%s:%d", externalIPAddresses[0],
+				functionNodePort)
+			i.createFunctionInvocationOptions.SkipURLValidation = true
+		} else {
+			return errors.New("Function has no node port and thus cannot be invoked externally " +
+				"while providing external ip addresses")
+		}
+		return nil
+	}
+
+	if len(invocationURLs) < 2 {
+		return errors.New("Function has no external invocation url")
+	}
+
+	// use last invocation url
+	// implementation detail: first url is the intra-cluster url, following urls are external urls
+	// the last url is the one that is most likely to be an in gress.
```

Comment:
```suggestion
	// the last url is the one that is most likely to be an ingress.
```
Most likely according to what?
And happens what if its not an ingress?

Reply:
node port, will add comment